File: ruby/lib/coding_interview/application/repository/account_repository.rb
module CodingInterview
  module Application
    module Repository
      # 口座管理リポジトリ。
      class AccountRepository
        def find(_user_id); raise NotImplementedError; end
        def upsert(_user_id, _account); raise NotImplementedError; end
        def exists?(_user_id); raise NotImplementedError; end
      end
    end
  end
end

File: python/src/coding_interview/application/repository/account_repository.py
from __future__ import annotations

from typing import Protocol

from coding_interview.domain.account import Account
from coding_interview.domain.user_id import UserId


class AccountRepository(Protocol):
    """口座管理リポジトリ。"""

    def find(self, user_id: UserId) -> Account | None: ...
    def upsert(self, user_id: UserId, account: Account) -> None: ...
    def exists(self, user_id: UserId) -> bool: ...
